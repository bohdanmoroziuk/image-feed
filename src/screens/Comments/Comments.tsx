import { FunctionComponent } from 'react';
import { SafeAreaView, ViewProps } from 'react-native';

import { Comment } from 'src/types';

import NavigationBar from 'src/components/NavigationBar';
import CommentInput from 'src/components/CommentInput';
import CommentList from 'src/components/CommentList';

export interface CommentsProps {
  style?: ViewProps['style'];
  comments: Comment[];
  onClose: () => void;
  onSubmitComment: (text: string) => void;
}

export const Comments: FunctionComponent<CommentsProps> = ({
  style,
  comments,
  onClose,
  onSubmitComment
}) => (
  <SafeAreaView style={style}>
    <NavigationBar
      title="Comments"
      closeButtonText="close"
      onClose={onClose}
    />
    <CommentInput
      placeholder="Leave a comment"
      onSubmit={onSubmitComment}
    />
    <CommentList items={comments} />
  </SafeAreaView>
);
