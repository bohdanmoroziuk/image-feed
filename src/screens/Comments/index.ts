import { Comments } from './Comments';

export type { CommentsProps } from './Comments';
export default Comments;
