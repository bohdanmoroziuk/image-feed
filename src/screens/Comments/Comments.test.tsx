import { render } from '@testing-library/react-native';

import { Comments } from './Comments';

describe('Comments', () => {
  it('renders successfully', () => {
    const { toJSON } = render(
      <Comments
        comments={[{ id: '1', text: 'Comment' }]}
        onClose={() => {}}
        onSubmitComment={() => {}}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});