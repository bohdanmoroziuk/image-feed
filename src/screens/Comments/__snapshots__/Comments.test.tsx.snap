// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments renders successfully 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "rgba(0, 0, 0, 0.1)",
        "borderBottomWidth": 0.5,
        "height": 40,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "justifyContent": "center",
          "left": 20,
          "opacity": 1,
          "position": "absolute",
          "top": 0,
        }
      }
    >
      <Text>
        close
      </Text>
    </View>
    <Text
      style={
        Object {
          "fontWeight": "500",
        }
      }
    >
      Comments
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "rgba(0, 0, 0, 0.1)",
        "borderBottomWidth": 0.5,
        "height": 60,
        "paddingHorizontal": 20,
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Leave a comment"
      style={
        Object {
          "flex": 1,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "borderBottomColor": "rgba(0, 0, 0, 0.05)",
            "borderBottomWidth": 0.5,
            "marginLeft": 20,
            "paddingRight": 20,
            "paddingVertical": 20,
          }
        }
      >
        <Text>
          Comment
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
